added in AppContext and audio player

franknoirot opened the pull request:
Resolves #1, #2, and #4.

## Additional Work

- added nav links (no rollover yet)
-got the audio player playing continuously while navigating by making the nav a global component.
- added animations to the music notes while the song is playing.

robpetrin commented:
Hey @franknoirot eager to look into where this failed a little more. Is the netlify org for neta centrally-controlled or with you? I think payment begins after you add your second person, so it's not needed to add me if the creds are Frank-specific. But I can't look into this more without that. :D Thanks!

franknoirot commented:
Hey @robpetrin sorry the checks that were failing were left over from the Sanity starter that I had the site set up with originally. The starter created two sites in Netlify, the frontend and the CMS, both of which got built when the repo was pushed to. Though I had removed the Sanity Studio code from the repo, the site was still configured to try to rebuild, which of course failed. I deleted that site out of Netlify and invited you to the organization in there.

robpetrin commented:
Oh my goodness this really might change everything huh

franknoirot commented on components/Navbar/Navbar.js:
```diff
@@ -1,19 +1,39 @@
                 <Phonograph />
-                <Image className={styles.monogram} width="160" height="160" src='/./img/monogram.svg' alt="L + G"/>
+                <Link href='/'>
+                    <a>
```
Interesting bit of syntax, the NextJS Link component requires an `<a>` tag within it to work.

robpetrin replied:
interesting -- vanilla React does not, right?

franknoirot commented on components/Phonograph/Phonograph.module.css:
```diff
@@ -44,6 +44,19 @@
+    animation: play-notes 3s steps(2, end) infinite;
+}
+
+.active  .notes path:nth-of-type(2) {
```
The notes all animate the same, just with a slight offset.

franknoirot commented on lib/AppContext.js:
```diff
@@ -0,0 +1,34 @@
+    return useContext(AppContext)
+}
+
+function appReducer(state, action) {
```
Read this [bible](https://reactjs.org/docs/hooks-reference.html) for more info on useReducer and useContext. Way overkill for this, but very useful for larger projects.

franknoirot reviewed (commented):
Just left some explainer comments.

robpetrin reviewed (approved):
Hi @franknoirot -- looks great! Excited to collaborate more on these, and good call on the review requirements for PRs to main. No red flags here -- merging. See #5 for a future idea for our prettier 💅  -- thanks!
